src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/CustomAttributeWrapper.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class CustomAttributeWrapper
+    {
+        internal ConstructorInfo constructorInfo;
+        internal byte[] binaryAttribute;
+
+        public CustomAttributeWrapper(ConstructorInfo constructorInfo, byte[] binaryAttribute)
+        {
+            this.constructorInfo = constructorInfo;
+            this.binaryAttribute = binaryAttribute;
+        }
+    }
+
+    internal struct CustomAttributeInfo
+    {
+        public ConstructorInfo _ctor;
+        public object?[] _ctorArgs;
+        public string[] _namedParamNames;
+        public object?[] _namedParamValues;
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2057:Unrecognized value passed to the parameter 'typeName' of method 'System.Type.GetType(String)'",
+            Justification = "The 'enumTypeName' only available at runtime")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicFields', 'DynamicallyAccessedMemberTypes.NonPublicFields' in call to 'System.Type.GetField(String, BindingFlags)'",
+            Justification = "Could not propagate attribute into 'ctor.DeclaringType' only available at runtime")]
+        internal static CustomAttributeInfo DecodeCustomAttribute(ConstructorInfo ctor, byte[] data)
+        {
+            int pos;
+            CustomAttributeInfo info = default;
+
+            // Prolog
+            if (data.Length < 2)
+                throw new InvalidOperationException(SR.Format(SR.InvalidOperation_InvalidCustomAttributeLength, data.Length));
```

Comment:
When these exceptions are thrown (here and elsewhere), is there any context so the caller can determine which attribute and parent class type the invalid length is on? Does .NET Framework do anything special here?

Reply:
> When these exceptions are thrown (here and elsewhere), is there any context so the caller can determine which attribute and parent class type the invalid length is on?

The custom attributes added one my one so I think it will be clear which attribute, I can improve the message to include it is about invalid `binaryData`, also looks its better to be ArgumentException